tests/conditional_compilation/test_infer.py

```diff
@@ -5,19 +5,34 @@
 
 """ Test inference with conditional compiled binaries.
 """
+
 import sys
+from inspect import getsourcefile
+from pathlib import Path
 from proc_utils import cmd_exec  # pylint: disable=import-error
 from install_pkg import get_openvino_environment  # pylint: disable=import-error
 
 
-def test_infer(test_id, model, artifacts):
+def run_infer(model, out, install_dir):
+    """ Function running inference
+    """
+
+    returncode, output = cmd_exec(
+        [sys.executable,
+         str((Path(getsourcefile(lambda: 0)) / ".." / "tools" / "infer_tool.py").resolve()),
+         "-d=CPU", f"-m={model}", f"-r={out}"
+         ],
+        env=get_openvino_environment(install_dir),
+    )
+    return returncode, output
+
+
+def test_infer(test_id, model, artifacts, openvino_root_dir):
     """ Test inference with conditional compiled binaries
     """
     install_prefix = artifacts / test_id / "install_pkg"
-    exe_suffix = ".exe" if sys.platform == "win32" else ""
-    benchmark_app = install_prefix / "bin" / f"benchmark_app{exe_suffix}"
-    returncode, _ = cmd_exec(
-        [str(benchmark_app), "-d=CPU", f"-m={model}", "-niter=1", "-nireq=1"],
-        env=get_openvino_environment(install_prefix),
-    )
-    assert returncode == 0, f"Command exited with non-zero status {returncode}"
+    out = artifacts / test_id
+    returncode, output = run_infer(model, out, openvino_root_dir)
+    assert returncode == 0, f"Command exited with non-zero status {returncode}:\n {output}"
```

Comment:
`test_infer` still should check minimized build works fine. No need to infer original build. For comparison with original there is `test_verify`